subgraphs/rocket-pool/src/entityUpdates/financialMetrics.ts

```diff
@@ -0,0 +1,440 @@
+import {
+  Address,
+  BigInt,
+  BigDecimal,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+import { bigIntToBigDecimal } from "../utils/numbers";
+import { getOrCreateProtocol } from "../entities/protocol";
+import { getOrCreatePool } from "../entities/pool";
+import {
+  FinancialsDailySnapshot,
+  Minipool,
+  PoolDailySnapshot,
+  PoolHourlySnapshot,
+} from "../../generated/schema";
+import {
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  ETH_ADDRESS,
+  RETH_ADDRESS,
+  SECONDS_PER_DAY,
+  SECONDS_PER_HOUR,
+  RPL_ADDRESS,
+  BIGDECIMAL_HALF,
+} from "../utils/constants";
+import { getOrCreateRewardToken, getOrCreateToken } from "../entities/token";
+import { getOrCreateMinipool } from "../entities/Minipool";
+
+const PROTOCOL_ID = RETH_ADDRESS;
+
+export function updateMinipoolTvlandRevenue(
+  block: ethereum.Block,
+  stakingAmount: BigInt,
+  slashingAmount: BigInt,
+  totalRewardEth: BigInt,
+  miniPoolAddress: string
+): void {
+  const Minipool = getOrCreateMinipool(
+    block.number,
+    block.timestamp,
+    miniPoolAddress
+  );
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  Minipool.inputTokenBalances = [
+    Minipool.inputTokenBalances[0].plus(stakingAmount),
+  ];
+
+  Minipool.slashAmount = Minipool.slashAmount.plus(slashingAmount);
```

Comment:
Slashed amounts should deduct from the `inputTokenBalances`?

Reply:
This is done at a later stage in data processing.